### ISSUE TITLE
After upgrade to  v0.107.19 - AGH stops responding sometimes, slow to restart

### ISSUE THREAD

**Issue by timkgh** [AdguardTeam/AdGuardHome#5213]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not ask a question

### Operating system type

Other (please mention in the description)

### CPU architecture

64-bit MIPS

### Installation

GitHub releases or script from README

### Setup

On a router, DHCP is handled by the router

### AdGuard Home version

 v0.107.19

### Description

v0.107.19 running on EdgeRouter ER-12 (mips64_softfloat)

After the upgrade to v0.107.19, AGH sometimes stops responding to all DNS requests (UDP 53). In this state it is very slow to even restart, `systemctl stop AdGuardHome` takes a good 18 seconds.

Something seems to have broken between v0.107.18 and v0.107.19

**Comment by nycom83** [comment]
I have the same issue

**Comment by antoinelibert** [comment]
Do you have Adguard browsing security web service enabled? I had AG stopping to respond to query randomly two times the last two weeks (but still showing a successful query in the web interface).

The first time, I restarted the container then the whole server and it was up again. The second time, I had restarted and downgraded to .17 with no avail. Only after I disabled the browsing WS that it worked again but I don't have anything that could prove that was it... 

![image](https://user-images.githubusercontent.com/10107072/204629824-bdf6c685-b082-4799-b541-4ae8ee8f486e.png)

**Comment by timkgh** [comment]
Yep, this is it! I was about to post it. AGH gets in a bad, unresponsive state when the AG DNS servers are not available.

```
Nov 29 11:51:38 router AdGuardHome[29585]: 2022/11/29 11:51:38.122893 [error] handling tcp: handling tcp request: talking to dns upstream: checking host "cnn.com": parental: requesting https://family.adguard-dns.com:443/dns-query: Get "https://family.adguard-dns.com:443/dns-query?dns=AAABAAABAAAAAAAABGFkOGECcGMDZG5zB2FkZ3VhcmQDY29tAAAQAAE": all dialers failed: 4 errors: "dial tcp 94.140.14.15:443: connect: connection refused", "dial tcp 94.140.15.16:443: connect: connection refused", "dial tcp [2a10:50c0::bad1:ff]:443: connect: cannot assign requested address", "dial tcp [2a10:50c0::bad2:ff]:443: connect: cannot assign requested address"

Nov 29 11:52:10 router AdGuardHome[29585]: 2022/11/29 11:52:10.807781 [error] handling tcp: handling tcp request: talking to dns upstream: checking host "cnn.com": parental: requesting https://family.adguard-dns.com:443/dns-query: Get "https://family.adguard-dns.com:443/dns-query?dns=AAABAAABAAAAAAAABGFkOGECcGMDZG5zB2FkZ3VhcmQDY29tAAAQAAE": all dialers failed: 4 errors: "dial tcp 94.140.14.15:443: connect: connection refused", "dial tcp 94.140.15.16:443: connect: connection refused", "dial tcp [2a10:50c0::bad1:ff]:443: connect: cannot assign requested address", "dial tcp [2a10:50c0::bad2:ff]:443: connect: cannot assign requested address"
```

**Comment by daryll-swer** [comment]
I have the same issue, but Safe Browsing is disabled on my end. It seems to be related to a new bug:
![image](https://user-images.githubusercontent.com/80836254/204638190-097b7d0c-f56b-4d9d-bd77-50704d2f0924.png)

**Comment by Kooramah** [comment]
I have the same issue, since I've upgraded. Web browsing is slow.

I though its a DNS issue but its not. I changed the DNS to something else other than adguard and then everything seemed normal.

Anything happened on AGDNS? Queries are taking over 8 seconds to resolve

**Comment by Kooramah** [comment]
> Do you have Adguard browsing security web service enabled? I had AG stopping to respond to query randomly two times the last two weeks (but still showing a successful query in the web interface). The last one was like 10-15 mins ago.
> 
> The first time, I restarted the container then the whole server and it was up again. The second time, I had restarted and downgraded to .17 with no avail. Only after I disabled the browsing WS that it worked again but I don't have anything that could prove that was it...
> 
> ![image](https://user-images.githubusercontent.com/10107072/204629824-bdf6c685-b082-4799-b541-4ae8ee8f486e.png)

I had this enabled, and I just disabled it and everything went back to normal...

May be the culprit

**Comment by timkgh** [comment]
It seems that the upstream AG DNS servers are having issues. If you change your upstreams DNS servers to something else, e.g. Google or Cloudflare and disable `Use AdGuard browsing security web service` and `Use AdGuard parental control web service` it seems to be back to normal.

**Comment by talesam** [comment]
I have the same problem in this version, after I updated, I've been testing all day to find out what it is, then I went to test another installation of adguard that I also updated and it has the same problem.

**Comment by talesam** [comment]
> Use AdGuard browsing security web service

That's not it yet, I'm going to downgrade Adguard, I disabled it and it still slows down.

**Comment by Churator** [comment]
Same here, DNS stop resolving, docker restart solve it temporarly.

**Comment by talesam** [comment]
Just downgrade to the previous version and it will work normally.

https://github.com/AdguardTeam/AdGuardHome/releases/tag/v0.107.18

**Comment by Gitucc** [comment]
I have same issues.
I don't enable neither Safe Browsing nor Parents Control. After several attempts, I found out that those issues come out after 108-b.19. In this picture, ADGH took 90 seconds to stop in my VPS when using version b.19 and newer , but b.18 didn't.
![IMG_20221201_214145_453](https://user-images.githubusercontent.com/54905593/205068144-af6c936e-bda6-4ef2-ad34-59c82dd02ced.jpg)
Besides, I also encounter DNS over TLS and DNS over QUIC brake everyday when using the version higher than b.18, but in the meantime DNS over HTTPS/3 are still working fine, after a time consuming restart everything goes fine till it brakes again.
So I stay using b.18, I tried reading verbose log, but I couldn't figure out what's going on.

**Comment by timkgh** [comment]
No, not with v0.108.0-b.54
